source/extensions/resource_monitors/fixed_heap/config.cc

```diff
@@ -0,0 +1,32 @@
+#include "extensions/resource_monitors/fixed_heap/config.h"
+
+#include "envoy/registry/registry.h"
+
+#include "common/protobuf/utility.h"
+
+#include "extensions/resource_monitors/fixed_heap/fixed_heap_monitor.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ResourceMonitors {
+namespace FixedHeapMonitor {
+
+Server::ResourceMonitorPtr FixedHeapMonitorFactory::createResourceMonitor(
+    const Protobuf::Message& config,
+    Server::Configuration::ResourceMonitorFactoryContext& /*unused_context*/) {
+  return std::make_unique<FixedHeapMonitor>(
+      MessageUtil::downcastAndValidate<
```

Comment:
To avoid the cast/validation boilerplate, some other extension mechanisms have a templated adapter to the extension interface. Search for "FromProtoTyped" to see examples of this. Not necessarily something needed in this PR, but could be good to add a TODO at least.